src/shared/date/date-text.js

```diff
@@ -0,0 +1,65 @@
+import { useConfig, useTimeZoneConversion } from '@dhis2/app-runtime'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { convertFromIso8601ToString } from './date-utils.js'
+
+const formatDate = ({
+    dateString,
+    dateFormat = 'yyyy-mm-dd',
+    includeTimeZone = false,
+}) => {
+    if (!dateString) {
+        return ''
+    }
+    // the returned date includes seconds/ms and we want to simplify to just show date and HH:MM
+    const year = dateString.substring(0, 4)
+    const month = dateString.substring(5, 7)
+    const day = dateString.substring(8, 10)
+    const minutes = dateString.substring(11, 13)
+    const seconds = dateString.substring(14, 16)
+
+    const timeZone = Intl.DateTimeFormat()?.resolvedOptions()?.timeZone
+
+    if (dateFormat.toLowerCase() === 'dd-mm-yyyy') {
+        return `${day}-${month}-${year} ${minutes}:${seconds} ${
+            includeTimeZone && timeZone ? '(' + timeZone + ')' : ''
+        }`
+    }
+    return `${year}-${month}-${day} ${minutes}:${seconds} ${
+        includeTimeZone && timeZone ? '(' + timeZone + ')' : ''
+    }`
+}
+
+export const DateText = ({ date, includeTimeZone }) => {
+    const { systemInfo = {} } = useConfig()
+    const { calendar = 'gregory', dateFormat } = systemInfo
+    const { fromServerDate } = useTimeZoneConversion()
+
+    // NOTE: the passed date is assumed to be in ISO
```

Comment:
Yeah. I don't know if there's much we can do if we pass the wrong date because an Ethiopian date like `2016-03-14` is also a valid Gregorian date. But, I've added a step to check that the date is parsable as ISO, so a date like `2016-13-04` will fail and then I've set it to return `Invalid date ({{date you passed}})`